lambda/create-case/uid_generator.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"crypto/rand"
+	"math"
+	"math/big"
+	"strconv"
+	"strings"
+)
+
+var AllChars = []string{"3", "4", "6", "7", "8", "9", "Q", "W", "E", "R", "T", "Y", "U", "P", "A", "D", "F", "G", "H", "J", "K", "L", "X", "C", "V", "B", "N", "M"}
+var N = float64(len(AllChars))
+
+func generateUID() (string, error) {
+	numChars := []string{"3", "4", "6", "7", "8", "9"}
+
+	uid := ""
+	lettersInSequence := 0
+
+	for x := 1; x <= 11; x++ {
+		max := len(AllChars)
+
+		if lettersInSequence == 2 {
+			max = len(numChars)
+		}
+
+		index, err := rand.Int(rand.Reader, big.NewInt(int64(max)))
+		if err != nil {
+			return "", err
+		}
+
+		i := index.Int64()
+		if _, err := strconv.Atoi(AllChars[i]); err != nil {
+			lettersInSequence++
+		} else {
+			lettersInSequence = 0
+		}
+
+		uid = uid + AllChars[i]
+	}
+
+	checksum := generateChecksum(uid)
+
+	return "MTEST" + uid + checksum, nil
```

Comment:
Should this include hyphens (e.g. `MTEST-XXXX-XXXX-XXXX`)?

Reply:
Ah yes it should, i'll add those in. On a separate note, i'm assuming we'll want to strip hyphens before storing in the DB right?

Reply:
Good question. I assumed that we would since I see the hyphens as part of it. But if we don't, then there wouldn't be any point hyphenating here.

Reply:
I think the only point of hyphenating is to make it easier for users to read & input. So we don't really care about them on the backend. I think storing UIDs without hyphens is slightly better, as we can give users flexibility in searching with or without hyphens, as in the backend we can always strip them out before perform a query. Arguably we could do the reverse of that and hyphenate inputs before querying, but doing it that way seems weird...

Reply:
You're right, I'll tweak that on mine so we output it hyphenated but don't save it that way 👍🏽